File: leetcode/bit-manipulation/0191_number_of_1_bits.py
# LC 191. Number of 1 Bits | Easy
class Solution:
    def hammingWeight_builtin(self, n): return bin(n).count('1')
    # Brian Kernighan
    def hammingWeight(self, n):
        count = 0
        while n: n &= n - 1; count += 1
        return count

File: leetcode/bit-manipulation/0191_number_of_1_bits.cpp
// LC 191. Number of 1 Bits | Easy
class Solution {
public:
    int hammingWeight(int n) { int c = 0; while (n) { n &= n-1; c++; } return c; }
    int hammingWeight_builtin(int n) { return __builtin_popcount(n); }
};
